client/src/app/pages/main/gamemodes/gamemodes.component.ts

```diff
@@ -94,48 +83,20 @@ export class GamemodesComponent implements OnInit, OnDestroy {
   ];
 
   currentGameMode?: GameModeDetails;
-  futureGameMode?: GameModeDetails;
-
-  constructor(public gallery: Gallery) { }
-
-  ngOnInit() {
-    // Map images for gallery
-    this.currentGamemodesItems = this.currentGameModes.map(
-        item => new CustomGalleryItem({ src: item.imageUrl, thumb: item.imageUrl, youtubeLink: item.url }),
-      );
-    this.futureGamemodesItems = this.futureGameModes.map(
-        item => new CustomGalleryItem({ src: item.imageUrl, thumb: item.imageUrl, youtubeLink: item.url }),
-      );
 
-    // Get gallery references
-    const currentGalleryRef = this.gallery.ref('current-gamemodes');
-    const futureGalleryRef = this.gallery.ref('future-gamemodes');
+  gameModeIndex: number;
 
-    // Init galleries
-    this.currentGameMode = this.currentGameModes[0];
-    this.futureGameMode = this.futureGameModes[0];
+  constructor() { }
 
-    // Subscribe to gallery index changed event
-    currentGalleryRef.indexChanged.pipe(takeUntil(this.ngUnsub)).subscribe((galleryState: GalleryState) => {
-        this.currentGameMode = this.currentGameModes[galleryState.currIndex];
-    });
-    futureGalleryRef.indexChanged.pipe(takeUntil(this.ngUnsub)).subscribe((galleryState: GalleryState) => {
-      this.futureGameMode = this.futureGameModes[galleryState.currIndex];
-    });
+  setCurrentGamemode(index: number) {
+    if (index < this.gameModes.length && index >= 0) {
+      this.currentGameMode = this.gameModes[index];
+      this.gameModeIndex = index;
+    }
   }
 
-  ngOnDestroy(): void {
-    this.ngUnsub.next();
-    this.ngUnsub.complete();
-  }
-}
-
-export class CustomGalleryItem implements GalleryItem {
-
-  readonly type = 'custom-item';
-  readonly data: any;
-
-  constructor(data: any) {
-    this.data = data;
+  ngOnInit() {
+    this.currentGameMode = this.gameModes[0];
+    this.gameModeIndex = 0;
```

Comment:
Since this is static information (already in this file), you can do this in the component's constructor and remove `ngOnInit`

Reply:
should be resolved in 4f58980